R/utility_spliceHelper_functions.R

```diff
@@ -20,220 +20,336 @@
-                                                      queryStartEnd.subjectSplice.IdMap,
-                                                      sum))
-
-  # exon 3' splice site
-
-  queryExonEnd <- start(querySplice.subjectFullQuerySplice.Rng)<end(subjectFullQuerySplice.Rng) & 
-    start(querySplice.subjectFullQuerySplice.Rng)>start(subjectFullQuerySplice.Rng) &
-    end(querySplice.subjectFullQuerySplice.Rng)>end(subjectFullQuerySplice.Rng)
-  queryExonStart <- end(querySplice.subjectFullQuerySplice.Rng)<end(subjectFullQuerySplice.Rng) & 
-    end(querySplice.subjectFullQuerySplice.Rng)>start(subjectFullQuerySplice.Rng) &
-    start(querySplice.subjectFullQuerySplice.Rng)<start(subjectFullQuerySplice.Rng)
-
-  exon5Prime <- tapply(queryExonStart, querySplice.subjectFullQuerySplice.IdMap, sum)
-  exon3Prime <- tapply(queryExonEnd, querySplice.subjectFullQuerySplice.IdMap, sum)
-
-  subjectStartExonStartExtension <- start(subjectStart.querySplice.Rng)<end(qSpRngList) & end(subjectStart.querySplice.Rng)>end(qSpRngList) & start(qSpRngList)< start(subjectStart.querySplice.Rng)
-  subjectStartExonStartExtension <- tapply(subjectStartExonStartExtension, subjectStartEnd.querySplice.IdMap, sum)
-
-  subjectStartExonEndExtension <- end(subjectStart.querySplice.Rng)>start(qSpRngList) & start(subjectStart.querySplice.Rng)<start(qSpRngList) & end(qSpRngList)> end(subjectStart.querySplice.Rng)
-  subjectStartExonEndExtension <- tapply(subjectStartExonEndExtension, subjectStartEnd.querySplice.IdMap, sum)
-
-  subjectEndExonStartExtension <- start(subjectEnd.querySplice.Rng)<end(qSpRngList) & end(subjectEnd.querySplice.Rng)>end(qSpRngList) & start(qSpRngList)< start(subjectEnd.querySplice.Rng)
-  subjectEndExonStartExtension <- tapply(subjectEndExonStartExtension, subjectStartEnd.querySplice.IdMap, sum)
-
-  subjectEndExonEndExtension <- end(subjectEnd.querySplice.Rng)>start(qSpRngList) & start(subjectEnd.querySplice.Rng)<start(qSpRngList) & end(qSpRngList)> end(subjectEnd.querySplice.Rng)
-  subjectEndExonEndExtension <- tapply(subjectEndExonEndExtension, subjectStartEnd.querySplice.IdMap, sum)
-
-  startEndTable$exon5Prime <- exon5Prime
-  startEndTable$exon5Prime[startEndTable$strand!='-'] <- startEndTable$exon5Prime[startEndTable$strand!='-'] - subjectStartExonStartExtension[startEndTable$strand!='-']
-  startEndTable$exon5Prime[startEndTable$strand=='-'] <- exon3Prime[startEndTable$strand=='-'] - subjectStartExonEndExtension[startEndTable$strand=='-']
-
-  startEndTable$exon3Prime <- exon3Prime
-  startEndTable$exon3Prime[startEndTable$strand!='-'] <- startEndTable$exon3Prime[startEndTable$strand!='-'] - subjectEndExonEndExtension[startEndTable$strand!='-']
-  startEndTable$exon3Prime[startEndTable$strand=='-'] <- exon5Prime[startEndTable$strand=='-'] - subjectEndExonStartExtension[startEndTable$strand=='-']
-  return(startEndTable)
+  strand <- as.character(getStrandFromGrList(query))
```

Comment:
Hi @yuukiiwa , given that the strand can be obtained with a single line, it actually brought up another question that I have, is it really necessary to have a function call `getStrandFromGrgList`?

Reply:
Hi @cying111, I do agree that the getStrandFromGrList function can be expressed as a command within the main compareTranscripts() function. The reason why we made this function was to make the main compareTranscripts() function cleaner to make it easier to be interpreted. If you think it is unnecessary I can replace the current function with the command.

Reply:
I think from the time efficient view as @jonathangoeke mentioned, then you can leave it as it is then!